cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -405,16 +407,30 @@ func newDeleteOptions() *metav1.DeleteOptions {
 	}
 }
 
+// UpdateClusterObjectEndpoint updates the status of a cluster API endpoint, clusterEndpoint
+// can be passed as hostname or hostname:port, if port is not present the default port 443 is applied.
 // TODO: Test this function
-func (c *client) UpdateClusterObjectEndpoint(controlPlaneIP, clusterName, namespace string) error {
+func (c *client) UpdateClusterObjectEndpoint(clusterEndpoint, clusterName, namespace string) error {
+	var endpointHost, endpointPort string
 	cluster, err := c.GetClusterObject(clusterName, namespace)
 	if err != nil {
 		return err
 	}
+	endpointHost, endpointPort, err = net.SplitHostPort(clusterEndpoint)
```

Comment:
Nit: shouldn't `endpointHost, endpointPort, err := net.SplitHostPort(clusterEndpoint)` just suffice? then you don't need to declare the var explicitly in line#414 above

Reply:
good catch!